### DISCUSSION TITLE
Rendering on half of the terminal

### DISCUSSION THREAD

**Discussion by jacek-kurlit** [ratatui/ratatui#983] [Discussion]
Hi,

I want to achieve same effect as on the screenshot.
I simply want to render my ratatui app on the half of the terminal window without affecting what is above.
Maybe there is option to EnterAlternateScreen but have it copied what is in current screen and render only on selected portion of the buffer?
![image](https://github.com/ratatui-org/ratatui/assets/6184283/8f225249-0064-4f42-8ba3-5173e5eeac29)
 Is this possible with ratatui? Every example I can see runs on entire terminal screen. I want to be able to change only little portion of the screen and clear it up after finishing.

**Comment by joshka** [comment]
Take a look at the inline example or at the atuin app.

**Comment by jacek-kurlit** [comment]
Thank you @joshka for the hint with inline example!

All I need to do was to create terminal with viewport
```
let mut terminal = Terminal::with_options(
    backend,
    TerminalOptions {
        viewport: Viewport::Inline(8),
    },
)?;
```

My guess is the default viewport is Fullscreen

**Reply by joshka** [reply]
yep
